### PR TITLE
chore: use ``node-fetch`` instead of ``request``

### LINKED ISSUE #1421: Replace request by node-fetch
Following https://github.com/asyncapi/cli/pull/1399 and my comment here:
https://github.com/asyncapi/cli/pull/1399#discussion_r1590794198

Use node-fetch and get rid of [request](https://www.npmjs.com/package/request) library (deprecated)

### PR DESCRIPTION
**Pull request by helios2003**
**Description**
Using `node-fetch` library instead of `request` since the latter is deprecated. <br />
**Related issue(s)**
Fixes #1421

### CODE REVIEW

File: scripts/fetch-asyncapi-example.js [hunk 1]
```diff
@@ -24,12 +24,20 @@ const TEMP_ZIP_NAME = 'spec-examples.zip';
 
 const fetchAsyncAPIExamplesFromExternalURL = () => {
   return new Promise((resolve, reject) => {
-    request(SPEC_EXAMPLES_ZIP_URL)
-      .pipe(fs.createWriteStream(TEMP_ZIP_NAME))
-      .on('close', () => {
-        console.log('Fetched ZIP file');
-        resolve();
-      }).on('error', reject);
+    fetch(SPEC_EXAMPLES_ZIP_URL)
```
Comment (Amzani) [inline scripts/fetch-asyncapi-example.js:27]: Error handling can be improved by wrapping by a try/catch block (see: https://github.com/node-fetch/node-fetch?tab=readme-ov-file#handling-exceptions)

File: scripts/fetch-asyncapi-example.js [hunk 2]
```diff
@@ -24,12 +24,20 @@ const TEMP_ZIP_NAME = 'spec-examples.zip';
...
+      .then((res) => {
+        if (res.status !== 200) {
+          reject(new Error(`Failed to fetch examples from ${SPEC_EXAMPLES_ZIP_URL}`));
+        }
+        const file = fs.createWriteStream(TEMP_ZIP_NAME);
+        res.body.pipe(file);
+        file.on('close', () => {
```
Comment (Amzani) [inline scripts/fetch-asyncapi-example.js:34]: Listen also the the `error` event
```
file.on('error', (err) => {
                    reject(err);
                });
```

**Review by Amzani — CHANGES_REQUESTED** [review]

**Comment by helios2003** [comment]
Done with the changes @Amzani

**Comment by helios2003** [comment]
Also, currently, we are using `node-fetch` v2.0.0 because we are using CommonJS syntax. The library only supports ES6 module syntax in the updated v3, so should we change it? (Even though they mention that critical bug fixes will be published for v2)
Ref: https://www.npmjs.com/package/node-fetch#commonjs

**Review by Amzani — APPROVED** [review]
LGTM

**Review by Shurtu-gal — APPROVED** [review]

**Pull request merged**